src/platforms/zephyr/CMakeLists.txt

```diff
@@ -0,0 +1,128 @@
+#
+# This file is part of AtomVM.
+#
+# Copyright 2023 Winford (Uncle Grumpy) <winford@object.stream>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+#
+
+cmake_minimum_required (VERSION 3.20)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(AtomVM LANGUAGES C)
+
+if (NOT BOARD)
+    message(FATAL_ERROR "No BOARD specified for device config generator")
+endif ()
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../../CMakeModules")
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
+
+# Options that make sense for this platform
+option(AVM_USE_32BIT_FLOAT "Use 32 bit floats." ON)
+option(AVM_VERBOSE_ABORT "Print module and line number on VM abort" OFF)
+option(AVM_CREATE_STACKTRACES "Create stacktraces" ON)
+option(AVM_NEWLIB_NANO "Use 'nano' newlib. Saves 46kB, no `long long` support" OFF)
+option(AVM_LOG_DISABLE "Disable log output" OFF)
+option(AVM_ENABLE_LOG_COLOR "Use color log output" OFF)
+option(AVM_ENABLE_LOG_LINES "Include source and line info for all enbled levels" OFF)
+option(AVM_CONFIG_REBOOT_ON_NOT_OK "Reboot when application exits with non 'ok' return" OFF)
+option(AVM_DISABLE_GPIO_NIFS "Disable GPIO nifs (input and output)" OFF)
+option(AVM_DISABLE_GPIO_PORT_DRIVER "Disable GPIO 'port' driver (input, output, and interrupts)" OFF)
+
+set(AVM_DISABLE_SMP ON FORCE)
```

Comment:
In the case of the Zephyr port, could testing for the `CONFIG_SMP` variable work? That variable is set in the Kconfig and made available in CMake.

See:
- https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_SMP
- https://github.com/zephyrproject-rtos/zephyr/blob/a6eef0ba3755f2530c5ce93524e5ac4f5be30194/cmake/modules/kconfig.cmake#L402

Reply:
Yes. I was realizing today that making KConfig options to set these options would be preferable to passing extra -D options to `west`.

Reply:
What do you think of structuring as a Zephyr module, in which the `CMakeLists.txt` could be something like:
```cmake
cmake_minimum_required (VERSION 3.20)

if(Zephyr_FOUND)
    set(AVM_PORT_ZEPHYR ON)
endif()

# Options that make sense for this platform
cmake_dependent_option(AVM_USE_32BIT_FLOAT "Use 32 bit floats." ON "NOT CONFIG_64BIT" OFF)
option(AVM_VERBOSE_ABORT "Print module and line number on VM abort" OFF)
option(AVM_CREATE_STACKTRACES "Create stacktraces" ON)
cmake_dependent_option(AVM_NEWLIB_NANO "Use 'nano' newlib. Saves 46kB, no `long long` support" ON "CONFIG_HAS_NEWLIB_LIBC_NANO" OFF)
option(AVM_LOG_DISABLE "Disable log output" OFF)
option(AVM_ENABLE_LOG_COLOR "Use color log output" OFF)
option(AVM_ENABLE_LOG_LINES "Include source and line info for all enbled levels" OFF)
option(AVM_CONFIG_REBOOT_ON_NOT_OK "Reboot when application exits with non 'ok' return" OFF)
option(AVM_DISABLE_GPIO_NIFS "Disable GPIO nifs (input and output)" OFF)
option(AVM_DISABLE_GPIO_PORT_DRIVER "Disable GPIO 'port' driver (input, output, and interrupts)" OFF)

include_directories(src/libAtomVM)  # libAtomVM
include_directories(src/platforms/zephyr)  # AtomVM (Zephyr port)
```

Reply:
Instead of implementing `main`, the port could provide an implementation called `avm_main` which the Zephyr application could call from its own `main` implementation.